tests/integration/conftest.py

```diff
@@ -0,0 +1,180 @@
+import json
+import os
+import shutil
+import subprocess
+import sys
+import time
+import pytest
+
+
+@pytest.fixture(scope="module")
+def root_dir(request):
+    yield request.path.parent.parent.parent
+
+
+@pytest.fixture(scope="module")
+def cli_path(root_dir):
+    yield root_dir / "bin" / "birdhouse"
+
+
+@pytest.fixture(scope="session", autouse=True)
+def check_stack_not_running(request):
+    """
+    Check that the stack isn't already running before starting the tests.
+    """
+    if request.config.getoption("--no-start-stack", None):
+        return
+    proc = subprocess.run(
+        "docker ps --filter label=com.docker.compose.project --format 'table {{.Label \"com.docker.compose.project\"}}'",
+        shell=True,
+        stdout=subprocess.PIPE,
+        universal_newlines=True,
+    )
+    if proc.returncode:
+        pytest.fail("Docker may not be running. This test cannot run without docker.")
+
+    test_project_name = os.getenv("TEST_COMPOSE_PROJECT_NAME", "birdhouse-test-stack")
+    project_name = os.getenv("COMPOSE_PROJECT_NAME", "birdhouse")
+    lines = proc.stdout.splitlines()
+    if test_project_name in lines or project_name in lines:
+        pytest.fail("Birdhouse is currently running. Please stop the software before running tests.")
+
+
+@pytest.fixture(scope="session")
+def tmp_data_persist_root(tmp_path_factory):
+    yield tmp_path_factory.mktemp("data-persist-root")
+
+
+@pytest.fixture(scope="session")
+def stack_info():
+    """
+    Used to store information about the birhouse stack start-up so that it
+    isn't started multiple times per session.
+    """
+    info = {"started": False}
+    yield info
+
+
+@pytest.fixture(scope="module", autouse=True)
+def load_stack_env(cli_path, local_env_file, stack_info):
+    if stack_info["started"]:
+        return
+    proc = subprocess.run(
+        f"{cli_path} -e {local_env_file} configs -c 'env -0'",
+        shell=True,
+        stderr=subprocess.PIPE,
+        stdout=subprocess.PIPE,
+        universal_newlines=True,
+    )
+    if proc.returncode:
+        pytest.fail(f"Unable to get environment variables. Error:\n{proc.stderr}")
+    env_vars = dict(line.split("=", 1) for line in proc.stdout.split("\x00") if "=" in line)
+    print(env_vars["MAGPIE_ADMIN_USERNAME"])
+    stack_info["env_vars"] = env_vars
+
+
+@pytest.fixture(scope="module")
+def stack_env(stack_info):
+    return stack_info["env_vars"]
+
+
+@pytest.fixture(scope="module")
+def birdhouse_url(stack_env):
+    return f"{stack_env['BIRDHOUSE_PROXY_SCHEME']}://{stack_env['BIRDHOUSE_FQDN_PUBLIC']}"
+
+
+@pytest.fixture(scope="module", autouse=True)
+def start_stack(request, cli_path, local_env_file, tmp_data_persist_root, stack_info, pytestconfig):
```

Comment:
I haven't seen usage of this new fixture.  In fact, same for most of the new fixtures here.

I am guessing you are simply setting up for the future?

Reply:
Fixtures with `autouse=True` are used automatically in tests within their scope. So you won't see most of these referenced explicitly in tests.